test/xUnit/csharp/test_Subsystem.cs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Management.Automation.Subsystem;
+using System.Threading;
+using Xunit;
+
+namespace PSTests.Sequential
+{
+    public static class SubsystemTests
+    {
+        private static readonly MyPredictor predictor1, predictor2;
+
+        static SubsystemTests()
+        {
+            predictor1 = MyPredictor.FastPredictor;
+            predictor2 = MyPredictor.SlowPredictor;
+        }
+
+        // This method needs to be updated when there are more than 1 subsystem defined.
+        private static void VerifySubsystemMetadata(SubsystemInfo ssInfo)
+        {
+            Assert.Equal(SubsystemKind.CommandPredictor, ssInfo.Kind);
+            Assert.Equal(typeof(ICommandPredictor), ssInfo.SubsystemType);
+            Assert.True(ssInfo.AllowUnregistration);
+            Assert.True(ssInfo.AllowMultipleRegistration);
+            Assert.Empty(ssInfo.RequiredCmdlets);
+            Assert.Empty(ssInfo.RequiredFunctions);
+        }
+
+        [Fact]
+        public static void GetSubsystemInfo()
+        {
+            SubsystemInfo ssInfo = SubsystemManager.GetSubsystemInfo(typeof(ICommandPredictor));
+
+            VerifySubsystemMetadata(ssInfo);
+            Assert.False(ssInfo.IsRegistered);
+            Assert.Empty(ssInfo.Implementations);
+
+            SubsystemInfo ssInfo2 = SubsystemManager.GetSubsystemInfo(SubsystemKind.CommandPredictor);
+            Assert.Same(ssInfo2, ssInfo);
+
+            ReadOnlyCollection<SubsystemInfo> ssInfos = SubsystemManager.GetAllSubsystemInfo();
+            Assert.Single(ssInfos);
```

Comment:
This will fail if we add new subsystem and we will have to fix this - why don't fix now?
Below too.

Reply:
Yes, it will fail when declaring a new subsystem, and it's expected. After adding a new one, this should be made `Assert.Equal(2, ssInfos.Count)`.